### PR TITLE
Bump version to 0.4.0

### PR DESCRIPTION
**Pull request by RaulPPelaez**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by RaulPPelaez** [comment]
@rusty1s is this version incompatible with CUDA 11.2?

**Comment by RaulPPelaez** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -15,16 +15,15 @@ package:
...
     - fix-cmake.patch
 
 build:
-  number: 4
+  number: 0
   string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
   string: cpu_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version == "None"]
   # Only build for CUDA >=11.2
-  skip: true  # [win or cuda_compiler_version in ("10.2", "11.0", "11.1")]
+  skip: true  # [win or cuda_compiler_version=="11.2"]
```
Comment (jakirkham) [inline recipe/meta.yaml:26]: Small style suggestion

```suggestion
  skip: true  # [win or cuda_compiler_version == "11.2"]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -40,11 +39,13 @@ requirements:
     - libgomp      # [linux]
     - parallel-hashmap
     - cutlass                        # [cuda_compiler_version not in (undefined, 'None')]
+    - cuda-version =={{ cuda_compiler_version }}  # [cuda_compiler_version not in (undefined, 'None')]
```
Comment (jakirkham) [inline recipe/meta.yaml:42]: We can skip this in `requirements/build` as `{{ compiler('cuda') }}` pulls this in

```suggestion
```

With `requirements/host` we still need it as there isn't another good way to provide it automatically

**Review by jakirkham — COMMENTED** [review]
Thanks Raul! 🙏

Had a couple minor suggestions below

File: recipe/meta.yaml [hunk 3]
```diff
@@ -15,16 +15,15 @@ package:
...
-
   patches:
     - fix-cmake.patch
 
 build:
-  number: 4
+  number: 0
   string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
   string: cpu_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version == "None"]
   # Only build for CUDA >=11.2
```
Comment (jakirkham) [inline recipe/meta.yaml:25]: Sorry missed this comment needs an update with the `skip` change

```suggestion
  # Only build for CUDA >=11.8
```

**Comment by jakirkham** [comment]
Looks good. Thanks Raul! 🙏

**Pull request merged**